protocol/Cargo.toml

```diff
@@ -10,15 +10,18 @@ rust-version = "1.63.0"
 
 [features]
 default = ["std"]
+# High-level wrappers using futures traits.
 async = ["std", "futures/std"]
+# High-level wrappers using tokio traits - may affect MSRV requirements.
 tokio = ["std", "tokio/io-util"]
 std = ["alloc", "bitcoin/std", "bitcoin_hashes/std", "chacha20-poly1305/std", "rand/std", "rand/std_rng"]
 alloc = ["chacha20-poly1305/alloc"]
 
 [dependencies]
 futures = { version = "0.3.30", default-features = false, optional = true }
-# Must be under 1.39.0 due to MSRV 1.63.0 requirement.
-tokio = { version = ">=1.37.0, <1.39.0", default-features = false, optional = true }
```

Comment:
Oh, yes, this was previously straight up wrong.

Reply:
Wrong as in, the consumer could always just override a the tokio version with a `patch` or something?

Reply:
Wrong that pinning belongs to `Cargo.lock` not `Cargo.toml`. Putting these into `Cargo.toml` causes huge problems with updates that would be otherwise trivial one-command things.

Reply:
Yea, using this lib downstream was what made me realize this is painful and not super productive. The intention wasn't to pin, but I forget the justification for the `>=1.37.0` floor, at least the `<1.39.0` ceiling is attempting to convey useful information to the consumer.

Reply:
The lower bound tends to be legitimate (but it's also often too strict) the upper one is the problem.